sle: Add salt test call to console_tests

okurz opened the pull request:
Add existing salt test module to sle main.pm. It can only work when the *asmm* module is activated and should succeed as soon as the *asmm* module includes salt.

One open question is why `lib/registration.pm` is pressing 'alt-n' two times. I had to disable the second call to enable asmm during graphical installation using (proxy-)SCC

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/os-autoinst/os-autoinst-distri-opensuse/1653)
<!-- Reviewable:end -->

okurz commented:
@mitiao @dzedro can you please a look as you should know how `lib/registration.pm` is intended to work

dzedro commented on lib/registration.pm:
```diff
@@ -102,13 +102,18 @@ sub fill_in_registration_data {
+                    wait_screen_change { assert_and_click "scc-module-$addon" };
+                    # don't confuse later upcoming dialogs with mouse cursor
+                    # near the middle of screen
+                    mouse_hide(1);
```
mouse should be already hidden in welcome.pm

dzedro commented on lib/registration.pm:
```diff
@@ -138,7 +143,11 @@ sub fill_in_registration_data {
+            # to add addons during installation this presses alt-n the
+            # second time and skip the next addon selection screen which
+            # would break tests
+            send_key 'alt-n' if check_var('VIDEOMODE', 'text');
```
Maybe it was wrong before, I don't know now
I just remember I was loosing my mind when worked on this registration and add-ons ...

mitiao replied:
This alt-n might work for 'next' after regcodes input.
I think it should be moved into above fill regcodes block if addons need codes.

okurz commented:
Updated:
* call `mouse_hide(1)` with this argument in welcome
* moved 'alt-n' into the block for registering addons with reg codes

I will verify locally again after the changes: lord.arch/tests/4894 and lord.arch/tests/4895

okurz commented:
updated:
* select module selection screen properly for new installer layout

Local verification runs:
 * http://lord.arch/tests/4909
 * http://lord.arch/tests/4908

Both fail for different reasons but they show that the module selection works

okurz commented:
@dgutu-suse please see this PR for suggested changes for module registration. Feel free to use the commits locally and see if you can make use of them :-)

dgutu-suse commented:
The only thing I don't see in screens is License screen. When I was registering against real SCC for every module you need to accept the license. I've checked the video and screenshots, I think this step which I'm mentioning is closer to real world user case.

okurz commented:
@dgutu-suse maybe the missing licenses is a problem coming from proxy-SCC vs. SCC? Can you crosscheck with SCC team?

sysrich commented:
@dgutu-suse ping - any luck answering @okurz's question?

coolo commented:
needs rebase

okurz commented:
salt is still not in ASMM for current sle dev version so we shouldn't merge this, closing.
